lms/services/assignment.py

```diff
@@ -82,11 +87,21 @@ def upsert_assignment(
             )
             self._db.add(assignment)
 
+            # For new assignments check if we are copying
+            # from an existing one on the current launch
+            historical_assignment = self.get_copied_from_assignment(lti_params)
+            if (
+                historical_assignment
+                and historical_assignment.extra.get("group_set_id")
+                and not extra.get("group_set_id")
+            ):
+                extra["group_set_id"] = historical_assignment.extra.get("group_set_id")
+
         assignment.document_url = document_url
         assignment.title = lti_params.get("resource_link_title")
         assignment.description = lti_params.get("resource_link_description")
         assignment.is_gradable = is_gradable
-        assignment.extra = extra if extra else assignment.extra or {}
+        assignment.extra = extra if extra else assignment.extra
```

Comment:
This might be a bit misleading because there's an unconditional assignment statement here but it actually doesn't always change the attribute. Might be clearer if just done the "normal" way:

```suggestion
        if extra:
            assignment.extra = extra
```